hlpr_feature_extraction/include/objectFeatures.hpp

```diff
@@ -36,46 +38,67 @@ class pc_cluster_features
 
 }
 
-  void boundingBoxWithZ(pcl::PointCloud<PointT> &cluster)
+  // Added by Priyanka - for non-planar segmentation
+  void orientedBoundingBox(pcl::PointCloud<PointT> &cluster)
   {
-    pcl::PointCloud<PointT>::Ptr projected_cluster = cluster.makeShared();
-    for(size_t j = 0; j < projected_cluster->points.size(); j++)
-      projected_cluster->points[j].z = 1;
-
-     volume2 = cluster.size();
-
-     pcl::getMinMax3D(cluster, min, max);
-     aligned_bounding_box = minAreaRect(projected_cluster);
-     aligned_bounding_box.center.z = min[2]+(max[2]-min[2])/2; //this should be maxheight/2+planeHeight
-     aligned_bounding_box.size.zSize = fabs((float)(max[2]-min[2])); //this should be max height!
-
-     aligned_bounding_box.fillQuatGivenAxisAngle();
-  }
+      pcl::PointCloud<PointT>::Ptr projected_cluster = cluster.makeShared();
+      volume2 = cluster.size();
+      pcl::getMinMax3D(cluster, min, max);
+      
+      std::vector <float> moment_of_inertia;
+      std::vector <float> eccentricity;
+      pcl::PointXYZ min_point_OBB;
+      pcl::PointXYZ max_point_OBB;
+      pcl::PointXYZ position_OBB;
+      Eigen::Matrix3f rotational_matrix_OBB;
+      //Eigen::Vector3f major_vector, middle_vector, minor_vector;
+
+      pcl::PointCloud<PointT>::Ptr cluster_xyzrgb (projected_cluster);
+      pcl::PointCloud<pcl::PointXYZ>::Ptr cloud(new pcl::PointCloud<pcl::PointXYZ>);
+      pcl::copyPointCloud(*cluster_xyzrgb, *cloud);
+      pcl::MomentOfInertiaEstimation <pcl::PointXYZ> feature_extractor;
+      feature_extractor.setInputCloud(cloud);
+      feature_extractor.compute();
+  
+      //feature_extractor.getMomentOfInertia (moment_of_inertia);
+      //feature_extractor.getEccentricity (eccentricity);
+      feature_extractor.getOBB(min_point_OBB, max_point_OBB, position_OBB, rotational_matrix_OBB);
+      //feature_extractor.getEigenValues(major_value, middle_value, minor_value);
+      //feature_extractor.getEigenVectors(major_vector, middle_vector, minor_vector);
+      //feature_extractor.getMassCenter(mass_center);
+
+      // Fill up the Box3D structure
+      Eigen::Vector3f position (position_OBB.x, position_OBB.y, position_OBB.z);
+      oriented_bounding_box.center.x = position(0);
+      oriented_bounding_box.center.y = position(1);
+      oriented_bounding_box.center.z = position(2);
+
+      std::cout<<"Center of OBB: ("<< position(0) <<", "<<position(1)<<", "<<position(2)<<")"<<std::endl; 
```

Comment:
I think this will spam output, right? should probably remove or comment out

Reply:
Yeah, delete.